src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -52,12 +52,21 @@ export class KernelProcess implements IKernelProcess {
     private get isPythonKernel(): boolean {
         return isPythonKernelConnection(this.kernelConnectionMetadata);
     }
+    public get canInterrupt() {
+        if (this.pythonDaemon) {
+            return true;
+        }
+        if (this._kernelConnectionMetadata.kernelSpec.interrupt_mode === 'message') {
+            return false;
```

Comment:
the raw kernel process can only interrupt pyton daemon kernels or kernsls that support interrupt messages